app/assets/javascripts/collections/_table.js

```diff
@@ -0,0 +1,103 @@
+
+(function(App) {
+
+  'use strict';
+
+  App.Collection.Table = Backbone.Collection.extend({
+
+    defaults: {
+      options: {
+        search: {
+          caseSensitive: false,
+          shouldSort: true,
+          tokenize: false,
+          threshold: 0.6,
+          location: 7,
+          distance: 100,
+          maxPatternLength: 32
+        }
+      }
+    },
+
+    initialize: function(models, settings) {
+      this.options = {
+        headers: settings.headers || {},
+        resultsPerPage: settings.resultsPerPage
+      };
+    },
+
+    _initSearch: function() {
+      var searchOptions = _.extend(this.defaults.options.search,
+        {keys: this._getSearchableHeaders() }),
+        list = this.toJSON();
+
+      this.fuse = new Fuse(list, searchOptions);
+
+      // this backup is to allow recover the collection in case
+      // the search returns 0 results
+      this.backupCollection = this.clone();
+    },
+
+    _getSearchableHeaders: function() {
+      var headers = this.options.headers;
+
+      return _.pluck(_.filter(headers, {searchable: true}), 'field_name');
+    },
+
+    paginate: function(page) {
+      var resultsPerPage = this.options.resultsPerPage,
+        start = resultsPerPage,
+        end = resultsPerPage;
+
+      start *= page;
+      end = start + resultsPerPage;
+
+      return this.toJSON().slice(start, end);
+    },
+
+    sortResults: function(comparator) {
+      var sortedResults = this.sortBy(function(model) {
```

Comment:
We could probably just use the `comparator` option of the collection instead of copying the data:
```javascript
function(comparator) {
  this.comparator = comparator;
  this.sort();
}
```